assets/wizards/advertising/views/placements/index.js

```diff
@@ -47,16 +51,122 @@ const getAdUnitsForSelect = adUnits => {
+		return ( sizes || [] ).find(
+			size => size[ 0 ] === sizeToCheck[ 0 ] && size[ 1 ] === sizeToCheck[ 1 ]
+		);
+	} );
+};
+
+/**
+ * Placement Control Component.
+ */
+const PlacementControl = ( {
+	adUnits = {},
+	bidders = {},
+	value = {},
+	disabled = false,
+	onChange,
+	...props
+} ) => {
+	const [ biddersErrors, setBiddersErrors ] = useState( {} );
+
+	useEffect( () => {
+		const errors = {};
+		Object.keys( bidders ).forEach( bidderKey => {
+			const bidder = bidders[ bidderKey ];
+			const supported =
+				value.ad_unit &&
+				adUnits[ value.ad_unit ] &&
+				hasAnySize( bidder.ad_sizes, adUnits[ value.ad_unit ].sizes );
+			errors[ bidderKey ] =
+				! value.ad_unit || ! adUnits[ value.ad_unit ] || supported
+					? null
+					: sprintf(
+							// Translators: Ad bidder name.
+							__( '%s does not support the selected ad unit sizes.', 'newspack' ),
+							bidder.name,
+							''
+					  );
+		} );
+		setBiddersErrors( errors );
+	}, [ adUnits, value.ad_unit ] );
+
+	return (
+		<Fragment>
+			<SelectControl
+				label={ __( 'Ad Unit', 'newspack' ) }
+				value={ value.ad_unit }
+				options={ getAdUnitsForSelect( adUnits ) }
+				onChange={ data => {
+					onChange( {
+						...value,
+						ad_unit: data,
+					} );
+				} }
+				disabled={ disabled }
+				{ ...props }
+			/>
+			{ Object.keys( bidders ).map( bidderKey => {
+				const bidder = bidders[ bidderKey ];
+				// Translators: Bidder name.
+				const bidderLabel = sprintf( __( '%s Placement ID', 'newspack' ), bidder.name );
+				return (
+					<TextControl
+						key={ bidderKey }
+						value={ value.bidders_ids ? value.bidders_ids[ bidderKey ] : null }
+						label={ bidderLabel }
+						disabled={ biddersErrors[ bidderKey ] || disabled }
+						onChange={ data => {
+							onChange( {
+								...value,
+								bidders_ids: {
+									...value.bidders_ids,
+									[ bidderKey ]: data,
+								},
+							} );
+						} }
+						{ ...props }
+					/>
+				);
+			} ) }
+			{ Object.keys( biddersErrors ).map( bidderKey => {
+				if ( biddersErrors[ bidderKey ] ) {
+					return (
+						<Notice key={ bidderKey } isWarning>
+							{ biddersErrors[ bidderKey ] }
+						</Notice>
+					);
+				}
+				return null;
+			} ) }
+		</Fragment>
+	);
+};
+
 /**
  * Advertising Placements management screen.
  */
 const Placements = ( { adUnits } ) => {
+	const [ initialized, setInitialized ] = useState( false );
 	const [ inFlight, setInFlight ] = useState( false );
 	const [ error, setError ] = useState( null );
+	const [ editingPlacement, setEditingPlacement ] = useState( null );
 	const [ placements, setPlacements ] = useState( {} );
-	const placementsApiFetch = options => {
-		setInFlight( true );
-		apiFetch( options )
+	const [ bidders, setBidders ] = useState( {} );
+	const [ biddersError, setBiddersError ] = useState( null );
+
+	const placementsApiFetch = async ( options, quiet = false ) => {
+		if ( ! quiet ) setInFlight( true );
+		await apiFetch( options )
```

Comment:
This is purely a non-blocking, subjective code-style thing, but what's the purpose of using `async`/`await` if still using promises syntax? I say either use promises or `async`/`await` with synchronous-style syntax, e.g.:

```js
const placementsApiFetch = async ( options, quiet = false ) => {
	if ( ! quiet ) setInFlight( true );
	try {
		const data = await apiFetch( options )
		setPlacements( data );
		setError( null );
	} catch( err ) {
		setError( err );
	} finally {
		if ( ! quiet ) setInFlight( false );
	}
};
```

Reply:
Good call! Done in 084b0cbce5c1577a39f85b55352730f95d3f3b45